bench/wtperf/wtperf.c

```diff
@@ -109,6 +120,16 @@ generate_key(CONFIG *cfg, char *key_buf, uint64_t keyno)
 	sprintf(key_buf, "%0*" PRIu64, cfg->key_sz - 1, keyno);
 }
 
+static inline uint64_t
+decode_key(CONFIG *cfg, char *key_buf)
+{
+	uint32_t i = 0;
+	for (;i < cfg->key_sz; i++)
+		if (key_buf[i] != 0)
+			break;
+	return (atoi (key_buf + i));
```

Comment:
Keys can be 64 bit unsigned values, so ```atoi``` is inadequate here. Some googling revealed ```strtoull``, which is more likely to be correct.